Get rid of the HdbSigInfo type.

dlacoste-esrf opened the pull request:
Replace it as much as possible with sigInfo.

Commit:
Get rid of the HdbSigInfo type.
Replace it as much as possible with sigInfo.
Code factorization in HdbData with the introduction of HdbDataArray and HdbDataScalar.
Override hashcode method in HdbSigInfo to allow prepared statements to be stored in a HashMap with HdbSigInfo as key.
Most of the static methods in HdbSigInfo should now be useless, but kept for compatibility reasons.

JeanLucPons commented:
Thanks for the job.
For me, the way the HdbSigInfo.type field (for backward compatibility ?) is updated is unclear ? getSigInfo() functions seems to not update it ?
Would it be possible to put all needed stuff for backward compatibility in a separate file, for instance having  HdbSigInfo which extends HdbSigInfoCompat where HdbSigInfoCompat contains the old stuff ?

dlacoste-esrf commented:
As it is, I removed completely the HdbSigInfo.type, so this version is not compatible.
I can make some additions to restore compatibility. The inheritance idea is quite good, didn't think of it, I'll see what I can do!

dlacoste-esrf committed:
dump java version to 1.8

dlacoste-esrf committed:
Create SignalInfo to replace HdbSigInfo.
Replace it everywhere it is used.
HdbSigInfo inherit from it and keep its type for backward compatibility.
HdbSigInfo type is set by constructor from parent SignalInfo.
HdbReader returns HdbSigInfo to other packages for backward compatibility
